torch/testing/_internal/common_distributed.py

```diff
@@ -877,6 +888,66 @@ def _check_return_codes(self, elapsed_time) -> None:
     def is_master(self) -> bool:
         return self.rank == 0
 
+# Utility base class for DDP based Multi Process Test cases
+# This abstracts the PG creation and deletion, the backends are selected based
+# on device type. The tests functions can be instantiated per device type using
+# common_device_type.instantiate_device_type_tests
+# other backends can add entry in backend() function
+class DistributedTestBase(MultiProcessTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self._spawn_processes()
+
+    def tearDown(self):
+        try:
+            os.remove(self.file_name)
+        except OSError:
+            pass
+
+    def backend(self, device) -> str:
+        if "cuda" in device:
+            return "nccl"
+        elif "hpu" in device :   # intel gaudi
+            return "hccl"
+        else :
+            return "gloo"
+
+    def device_handle(self, device) :
+        device_name = device.split(":")[0]
+        if hasattr(torch, device_name):
+            return getattr(torch, device_name, None)
+        else:
+            raise RuntimeError(
+                f"Device {device} library not found"
+            )
+
+    def create_pg(self, device):
+        num_visible_devices = self.device_handle(device).device_count()
+        store = torch.distributed.FileStore(self.file_name, num_visible_devices)
+        torch.distributed.init_process_group(
+            backend=self.backend(device),
+            world_size=self.world_size,
+            rank=self.rank,
+            store=store
+        )
+        if "nccl" in self.backend(device):
+            torch.cuda.set_device(self.rank)
+        return torch.distributed.distributed_c10d._get_default_group()
+
+    def rank_to_device(self, device):
+        num_visible_devices = self.device_handle(device).device_count()
+        visible_device = range(num_visible_devices)
+        num_devices_per_process = 1
+        rank_to_device = {
+            i : list(visible_device[i * num_devices_per_process : (i + 1) * num_devices_per_process])
+            for i in range(num_visible_devices)
+        }
+        return rank_to_device
```

Comment:
c10d have changed to support 1 device per process (rank) only. I feel that this logic can be simplified under that assumption.

Reply:
e.g. as simple as `rank % num_visible_devices`

Reply:
This makes a lot of sense 

I have applied the changes in: 67b6adbb78be2c2c6cebdf8005305c4ce1798830

Thanks